substra/sdk/rest_client.py

```diff
@@ -134,10 +134,14 @@ def _request(self, request_name, url, **request_kwargs):
             if e.response.status_code == 500:
                 raise exceptions.InternalServerError.from_request_exception(e)
 
+            if e.response.status_code in [502, 503, 504]:
+                raise exceptions.GatewayTimeout.from_request_exception(e)
+
             raise exceptions.HTTPError.from_request_exception(e)
 
         return r
 
+    @utils.retry_on_exception(exceptions=(exceptions.GatewayTimeout))
```

Comment:
You may also want to add a retry on `exceptions.Timeout` that is being raised when requests raises its own `Timeout`.

Reply:
That's a good comment!

IMO we don't want to retry on Timeout: in case of timeout we don't know if the request has been processed by the server and a retry might not be the good solution.

We want to retry when the server could not receive and process the request.

Maybe the error naming is misleading? GatewayError seems better.

502: BadGateway
503: ServiceUnavailable
504: GatewayTimeout

Maybe we should exclude 504? Or have dedicated exceptions. It could help for debugging purposes.

Reply:
as discussed, I've updated exception name here fc3473e